recipe/install_libclang.bat

```diff
@@ -1,11 +1,23 @@
-cd %SRC_DIR%\build
+@echo on
+
+cd %SRC_DIR%\clang\build
 ninja install
+if %ERRORLEVEL% neq 0 exit 1
+
 cd %LIBRARY_PREFIX%
 rmdir /s /q lib libexec share include
 
 move bin bin2
-
 mkdir bin
 
-move bin2\libclang.dll bin\
+setlocal enabledelayedexpansion
+if "%PKG_NAME%"=="libclang" (
+    REM unversioned
+    move bin2\libclang.dll bin\
```

Comment:
In unix, we need the unversioned shared object `libclang.so`, because we need to be able to link to the shared library using just `-lclang`. The shared library itself tells the linker that the downstream object should have a runtime dependence on `libclang.so.13`.

Windows is slightly different. When we have `-llibclang`, the linker tries to find `libclang.lib` (not `libclang.dll`).
Then `libclang.lib` tells the linker that the downstream object should have a runtime dependence on `libclang.dll` or whatever it was configured to say. We need `libclang.lib` to say that it needs `libclang-13.dll` instead and also generate that. We don't need a `libclang.dll`.
To do this, do something like,
```
if(WIN32)
    set_target_properties(libclang PROPERTIES RUNTIME_OUTPUT_NAME "libclang-${LLVM_VERSION_MAJOR}")
endif()
```
This generates `libclang.lib` and `libclang-13.dll` and the former refers to the latter.

Reply:
OK, thank you very much for taking the time to explain.

How should `libclang.lib` / `libclang-13.dll` be split across the versioned / unversioned output? It seems to me they need to go together into `libclang13`, and that `libclang` just depends on `libclang13` but has no additional symlinks / libraries?

Reply:
`libclang-13.dll` should be in the versioned output just like `libclang.so.13`
`libclang.lib` should in non-versioned output just like `libclang.so`